src/NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/FirstOrder/FirstOrderSchemeLts.hpp

```diff
@@ -0,0 +1,167 @@
+  using magnitude_of_face_normal_tag =
+      ::Tags::Magnitude<domain::Tags::UnnormalizedFaceNormal<Dim>>;
+  using argument_tags = tmpl::list<NumericalFluxComputerTag, n_dot_fluxes_tag,
+                                   magnitude_of_face_normal_tag, ArgsTags...>;
+  using volume_tags = tmpl::append<tmpl::list<NumericalFluxComputerTag>,
+                                   get_volume_tags<NumericalFlux>>;
+  static auto apply(
+      const NumericalFlux& numerical_flux_computer,
+      const db::const_item_type<n_dot_fluxes_tag>& normal_dot_fluxes,
+      const Scalar<DataVector>& face_normal_magnitude,
+      const db::const_item_type<ArgsTags>&... args) noexcept {
+    BoundaryData boundary_data{normal_dot_fluxes.number_of_grid_points()};
+    boundary_data.field_data.assign_subset(normal_dot_fluxes);
+    dg::NumericalFluxes::package_data(make_not_null(&boundary_data),
+                                      numerical_flux_computer, args...);
+    get<magnitude_of_face_normal_tag>(boundary_data.extra_data) =
+        face_normal_magnitude;
+    return boundary_data;
+  }
+};
+
+}  // namespace detail
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief Boundary contributions for a first-order DG scheme with local
+ * time-stepping
+ *
+ * This class is the local time-stepping equivalent to the
+ * `dg::FirstOrderScheme::FirstOrderScheme`. Notable differences are:
+ *
+ * - Boundary contributions are added to the `VariablesTag` directly.
+ * - We use the `Tags::BoundaryHistory` on mortars.
+ * - We need to store the face-normal magnitude in the boundary history, as
+ * opposed to the non-LTS case where we can retrieve it when needed.
+ */
+template <size_t Dim, typename VariablesTag, typename NumericalFluxComputerTag,
+          typename TemporalIdTag, typename TimeStepperTag>
+struct FirstOrderSchemeLts {
+ private:
+  using base = FirstOrderScheme<Dim, VariablesTag, NumericalFluxComputerTag,
+                                TemporalIdTag>;
+
+ public:
+  static constexpr size_t volume_dim = Dim;
+  using variables_tag = VariablesTag;
+  using numerical_flux_computer_tag = NumericalFluxComputerTag;
+  using NumericalFlux = db::const_item_type<NumericalFluxComputerTag>;
+  using temporal_id_tag = TemporalIdTag;
+  using receive_temporal_id_tag = ::Tags::Next<temporal_id_tag>;
+  using time_stepper_tag = TimeStepperTag;
+
+  using magnitude_of_face_normal_tag =
+      ::Tags::Magnitude<domain::Tags::UnnormalizedFaceNormal<volume_dim>>;
+  using BoundaryData = dg::SimpleBoundaryData<
+      typename base::BoundaryData::field_tags,
+      tmpl::push_back<typename base::BoundaryData::extra_data_tags,
+                      magnitude_of_face_normal_tag>>;
+  using boundary_data_computer = detail::boundary_data_computer_lts_impl<
+      volume_dim, variables_tag, numerical_flux_computer_tag, BoundaryData>;
+
+  using mortar_data_tag =
+      Tags::BoundaryHistory<BoundaryData, BoundaryData,
```

Comment:
Why are we storing (and sending) the magnitude of the remote face normal?

Reply:
We need to store it for the LTS coupling function, or can we get around that? Since we don't distinguish between local and remote data anymore this means it is also being communicated (only for LTS). If you think that's significant overhead that needs to be avoided I can make the code a bit more complicated and factor out local data that's not being projected from the mortar data.

Reply:
It's probably fine.  We can change it later if it becomes an issue.